app/services/cognito/forgot_password/validate_reset_password_jwt.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+##
+# Module used to wrap communication with Amazon Cognito
+module Cognito
+  ##
+  # Module used to manage the password change
+  module ForgotPassword
+    ##
+    # Class used for JWT validation.
+    class ValidateResetPasswordJwt < BaseService
+      ##
+      # Initializer method
+      #
+      # ==== Attributes
+      #
+      # +token+ - JWT token
+      #
+      def initialize(token:)
+        @token = token
+      end
+
+      ##
+      # Perform the call to Cognito service to get user attributes
+      def call
+        validate_token
+        update_token_expiration
+        token_data['email']
+      end
+
+      private
+
+      # Variable used internally by the service
+      attr_reader :token
+
+      # Token validation
+      def validate_token
+        if token_data['pw-reset-requested'].present? && extract_attr('custom:pw-reset-requested').present? &&
+           (extract_attr('custom:pw-reset-requested') == token_data['pw-reset-requested']) && last_request_1_day_ago?
```

Comment:
```suggestion
        if (extract_attr('custom:pw-reset-requested') == token_data['pw-reset-requested']) && last_request_1_day_ago?
```
please check if it can be in this way, i mean no nil exception, maybe add some tests to cover such cases